syntax = "proto3";

service RoverService {
  rpc GetMap(Empty) returns (MapData);
  rpc GetCommandStream(Empty) returns (stream Command);
  rpc GetMineSerialNumber(MineLocation) returns (MineSerial);
  rpc ReportCommandExecutionStatus(CommandStatus) returns (Empty);
  rpc ShareMinePIN(MinePIN) returns (Empty);
}

message Empty {}

message MapData {
  string map = 1;
}

message Command {
  string command = 1;
}

message MineLocation {
  int32 x = 1;
  int32 y = 2;
}

message MineSerial {
  string serial_number = 1;
}

message CommandStatus {
  bool success = 1;
  string message = 2;
}

message MinePIN {
  string pin = 1;
}
